include/drivers/adp536x.h

```diff
@@ -4,18 +4,183 @@
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_init(const char *dev_name);
+
+/**
+ * @brief Set the VBUS current limit.
+ *
+ * @param[in] value The upper current threshold in LSB.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_vbus_current_set(u8_t value);
+
+/**
+ * @brief Set the charger current.
+ *
+ * @param[in] value The charger current in LSB.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_current_set(u8_t value);
+
+/**
+ * @brief Set the charger termination voltage.
+ *
+ * This function configures the maximum battery voltage where
+ * the charger remains active.
+ *
+ * @param[in] value The charger termination voltage.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_termination_voltage_set(u8_t value);
+
+/**
+ * @brief Enable charging.
+ *
+ * @param[in] enable The requested charger operation state.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charging_enable(bool enable);
+
+/**
+ * @brief Read the STATUS1 register.
+ *
+ * @param[out] buf The read value of the STATUS1 register.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_status_1_read(u8_t *buf);
+
+/**
+ * @brief Read the STATUS2 register.
+ *
+ * @param[out] buf The read value of the STATUS2 register.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_charger_status_2_read(u8_t *buf);
+
+/**
+ * @brief Enable charge hiccup protection mode.
+ *
+ * @param[in] enable The requested hiccup protection state.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_oc_chg_hiccup_set(bool enable);
+
+/**
+ * @brief Enable discharge hiccup protection mode.
+ *
+ * @param[in] enable The requested hiccup protection state.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_oc_dis_hiccup_set(bool enable);
+
+/**
+ * @brief Enable the buck/boost regulator.
+ *
+ * @param[in] enable The requested regulator operation state.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_buckbst_enable(bool enable);
+
+/**
+ * @brief Set the buck regulator to 1.8 V.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
 int adp536x_buck_1v8_set(void);
+
+/**
+ * @brief Set the boost regulator to 3.3 V.
```

Comment:
Comment says `boost`, while the function name says buck-boost

Reply:
Fixed.